portfolio/src/main/java/com/google/sps/servlets/PostServlet.java

```diff
@@ -0,0 +1,96 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.BlogPost;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/post")
+public class PostServlet extends HttpServlet {
+
+  private static final String POST_PARAM = "blogpost";
+  private static final String POSTIN_PARAM = "blog-input";
+  private static final String TITLE_PARAM = "title";
+  private static final String TIME_PARAM = "timestamp";
+  private static final String CAT_PARAM = "category";
+  private static final String TAG_PARAM = "tag";
+  private static final String BLOGPOST = "BlogPost";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Query query = new Query(BLOGPOST).addSort(TIME_PARAM, SortDirection.ASCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    String requestTag = request.getParameter(TAG_PARAM);
+    List<BlogPost> posts = new ArrayList<>();
+
+    if (requestTag == null) {
+      for (Entity entity : results.asIterable()) {
+        String tag = (String) entity.getProperty(TAG_PARAM);
+        String title = (String) entity.getProperty(TITLE_PARAM);
+        String category = (String) entity.getProperty(CAT_PARAM);
+        String blogpost = (String) entity.getProperty(POST_PARAM);
+        long timestamp = (long) entity.getProperty(TIME_PARAM);
+        BlogPost post = new BlogPost(title, tag, category, blogpost, timestamp);
+        posts.add(post);
+      }
+    }
+    else {
+      for (Entity entity : results.asIterable()) {
+        String tag = (String) entity.getProperty(TAG_PARAM);
+        if (requestTag.equals(tag)) {
```

Comment:
Please use a filter instead of doing manual filtering. The database will be much more efficient at it.